### PR TITLE
Add support for open_for_read_only APIs

### PR DESCRIPTION
**Pull request by wqfish**
This change exposes the C APIs that open RocksDB for read only.

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -76,11 +91,51 @@ impl DB {
...
+            opts,
+            path,
+            cfs,
+            RWOption::ReadOnly {
+                error_if_log_file_exist,
+            },
+        )
     }
 
     /// Open a database with the given database options and column family descriptors.
```
Comment (aleksuss) [inline src/db.rs:123]:
```suggestion
    /// Opens a database with the given database options and column family descriptors.
```

File: src/db.rs [hunk 2]
```diff
@@ -50,6 +50,12 @@ unsafe impl Send for DB {}
 // use within the rocksdb library is generally behind a const reference
 unsafe impl Sync for DB {}
 
+// Option that specifies whether open DB for read only.
+enum RWOption {
```
Comment (aleksuss) [inline src/db.rs:54]: Does it make sense to add derives Clone and Copy here ?

Comment (aleksuss) [reply src/db.rs:54]: And I'd rename it to `AccessType`.

Comment (wqfish) [reply src/db.rs:54]: > Does it make sense to add derives Clone and Copy here ?

Probably not necessary right now. It's not a public type so we can always add later if needed. I'll rename it.

**Comment by calavera** [comment]
@aleksuss I have implemented `OpenAsSecondary` on top of the AccessType enum introduced in this pull request. I was wonder when you're going to merge it, so I can open a new pull request without complicating the review more than necessary.

**Comment by aleksuss** [comment]
@calavera I'm waiting for your PR. I think to include it in the 0.14.0 release.

**Pull request merged**